api/test-context/src/test_context.rs

```diff
@@ -296,8 +295,15 @@ impl TestContext {
         TransactionFactory::new(self.context.chain_id())
     }
 
-    pub fn root_account(&self) -> LocalAccount {
-        LocalAccount::new(aptos_test_root_address(), self.root_key.private_key(), 0)
+    pub async fn root_account(&self) -> LocalAccount {
```

Comment:
Why not just avoid fetching the sequence number altogether by storing it in the test_context? This would avoid having to make this root_account function async as well.

Reply:
but in this way we have to make new_text_context() an async func. I think it's fine either way.